src/main/java/org/opensearch/knn/plugin/transport/DeleteModelTransportAction.java

```diff
@@ -14,24 +14,33 @@
 import org.opensearch.action.ActionListener;
 import org.opensearch.action.support.ActionFilters;
 import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.client.Client;
 import org.opensearch.common.inject.Inject;
+import org.opensearch.common.util.concurrent.ThreadContext;
 import org.opensearch.knn.indices.ModelDao;
 import org.opensearch.tasks.Task;
 import org.opensearch.transport.TransportService;
 
 public class DeleteModelTransportAction extends HandledTransportAction<DeleteModelRequest, DeleteModelResponse> {
 
     private final ModelDao modelDao;
+    private final Client client;
 
     @Inject
-    public DeleteModelTransportAction(TransportService transportService, ActionFilters filters) {
+    public DeleteModelTransportAction(TransportService transportService, ActionFilters filters, Client client) {
         super(DeleteModelAction.NAME, transportService, filters, DeleteModelRequest::new);
         this.modelDao = ModelDao.OpenSearchKNNModelDao.getInstance();
+        this.client = client;
     }
 
     @Override
     protected void doExecute(Task task, DeleteModelRequest request, ActionListener<DeleteModelResponse> listener) {
-        String modelID = request.getModelID();
-        modelDao.delete(modelID, listener);
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            String modelID = request.getModelID();
+            modelDao.delete(modelID, listener);
+        } catch (Exception e) {
+            logger.error(e);
```

Comment:
Add the error message here for better debugging

Reply:
ack